qemu/tests/balloon_check.py

```diff
@@ -52,10 +52,13 @@ def get_ballooned_memory(self):
         :rtype: int
         """
         try:
-            output = self.vm.monitor.info("balloon")
-            ballooned_mem = int(re.findall(r"\d+", str(output))[0])
-            if self.vm.monitor.protocol == "qmp":
-                ballooned_mem = ballooned_mem / (1024 ** 2)
+            if self.vm.monitor:
+                output = self.vm.monitor.info("balloon")
+                ballooned_mem = int(re.findall(r"\d+", str(output))[0])
+                if self.vm.monitor.protocol == "qmp":
+                    ballooned_mem = ballooned_mem / (1024 ** 2)
+            else:
+                return self.get_ballooned_memory()
```

Comment:
I think we don't need to re-call `get_ballooned_memory` by itself, I maybe got your point to run it to get the vm.monitor until it is not None, but this will introduce another issue: `RecursionError: maximum recursion depth exceeded`

Reply:
modified